internal/martian/proxy.go

```diff
@@ -822,7 +822,20 @@ func (p *Proxy) connectHTTP(req *http.Request, proxyURL *url.URL) (res *http.Res
 	} else {
 		d = dialvia.HTTPProxy(p.dial, proxyURL)
 	}
-	d.ConnectRequestModifier = p.ConnectRequestModifier
+	d.ConnectRequestModifier = func(r *http.Request) error {
```

Comment:
This does not belong here.

Reply:
I know, but you need to somehow get the trace ID and CONNECT request won't have it in its context. That's why it's here.

Reply:
I think it does not belong to Marian but Forwarder and should be implemented there.
There should be zero changes to martian code.